src/blackmisc/aviation/flightplanaircraftinfo.cpp

```diff
@@ -0,0 +1,388 @@
+        {
+            s = "H/";
+        }
+        else if (m_wtc.isCategory(CWakeTurbulenceCategory::SUPER))
+        {
+            s = "J/";
+        }
+        s += m_aircraftIcao.getDesignator() % "/" % equipmentToFaaCode(m_comNavEquipment, m_ssrEquipment);
+        return s;
+    }
+
+    CAircraftIcaoCode CFlightPlanAircraftInfo::getAircraftIcao() const
+    {
+        return m_aircraftIcao;
+    }
+
+    CComNavEquipment CFlightPlanAircraftInfo::getComNavEquipment() const
+    {
+        return m_comNavEquipment;
+    }
+
+    CSsrEquipment CFlightPlanAircraftInfo::getSsrEquipment() const
+    {
+        return m_ssrEquipment;
+    }
+
+    CWakeTurbulenceCategory CFlightPlanAircraftInfo::getWtc() const
+    {
+        return m_wtc;
+    }
+
+    QString CFlightPlanAircraftInfo::convertToQString(bool) const
+    {
+        return asIcaoString();
+    }
+
+    void CFlightPlanAircraftInfo::parseIcaoEquipmentCode(const QString &equipment)
+    {
+        // Example: B789/H-SDE1E2E3FGHIJ2J3J4J5M1RWXY/LB1D1
+        QStringList firstSplit = equipment.split("/");
+        Q_ASSERT_X(firstSplit.size() == 3, Q_FUNC_INFO, "Cannot split string as required for the ICAO format");
+        if (!CAircraftIcaoCode::isValidDesignator(firstSplit[0]) || firstSplit[1].isEmpty() || firstSplit[2].isEmpty())
+        {
+            return; // Invalid equipment code, leave everything default initialized
+        }
+
+        m_aircraftIcao = CAircraftIcaoCode(firstSplit[0]);
+
+        try
+        {
+            m_ssrEquipment = CSsrEquipment(firstSplit[2]);
+        }
+        catch (const std::invalid_argument &)
+        {
+            m_ssrEquipment = CSsrEquipment();
+        }
+
+        QStringList secondSplit = firstSplit[1].split("-");
+        if (secondSplit.size() != 2)
+        {
+            return; // Invalid code, leave everything else default initialized
+        }
+
+        if (!secondSplit[0].isEmpty())
+        {
+            try
+            {
+                // if the wake turbulence category incorrectly contains more than one letter
+                // just take the first letter
+                m_wtc = CWakeTurbulenceCategory(secondSplit[0].at(0));
+            }
+            catch (std::invalid_argument &)
+            {
+                m_wtc = CWakeTurbulenceCategory();
+            }
+        }
+        else
+        {
+            m_wtc = CWakeTurbulenceCategory();
+        }
+
+        try
+        {
+            m_comNavEquipment = CComNavEquipment(secondSplit[1]);
+        }
+        catch (std::runtime_error &)
+        {
+            m_comNavEquipment = CComNavEquipment();
+        }
+    }
+
+    void CFlightPlanAircraftInfo::parseFaaEquipmentCode(const QString &equipment)
+    {
+        // Example: H/A346/L
+        QStringList split = equipment.split('/');
+        Q_ASSERT_X(split.size() == 2 || split.size() == 3, Q_FUNC_INFO, "Cannot split string as required for the FAA format");
```

Comment:
- [ ] Is it correct that this should be an assert? Do we have a separate function for validating input?

Reply:
This is checked before calling the method here:

https://github.com/swift-project/pilotclient/blob/5fc492bc8d576e66f92a24e631c072a340a1dd7a/src/blackmisc/aviation/flightplanaircraftinfo.cpp#L25-L28

And as the method is ``private``, it is only used internally.